File: SCALE.py
n, x = map(int, input().split())
if 2 * x > (3 ** n) - 1:
	print(-1)
else:
	ls = []
	while x:
		ls.append(x % 3)
		x //= 3
	ls.append(0)
	for i in range(len(ls)-1):
		if ls[i] == 2:
			ls[i+1] += 1
			ls[i] = -1
		elif ls[i] == 3:
			ls[i+1] += 1
			ls[i] = 0
	for i in range(len(ls)):
		if ls[i] == -1:
			print(i+1, end = " ")
	print()
	for i in range(len(ls)):
		if ls[i] == 1:
			print(i+1, end = " ")
	print()

File: SCALE.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl "\n"

int main(void) {
	ios::sync_with_stdio(false);
	cin.tie(0);
	cout.tie(0);

#ifndef ONLINE_JUDGE
	freopen("input.txt", "r", stdin);
//	freopen("output.txt", "w", stdout);
#endif

	long long n, x;
	cin >> n >> x;
	if (2 * x > pow(3, n) - 1)
		cout << -1 << endl;
	else {
		vector<int> v;
		while (x) {
			v.push_back(x % 3);
			x /= 3;
		}
		v.push_back(0);
		for (int i = 0; i < v.size()-1; i++) {
			if (v[i] == 2)
				v[i+1]++, v[i] = -1;
			else if (v[i] == 3)
				v[i+1]++, v[i] = 0;
		}
		for (int i = 0; i < v.size(); i++)
			if (v[i] == -1)
				cout << i + 1 << " ";
		cout << endl;
		for (int i = 0; i < v.size(); i++)
			if (v[i] == 1)
				cout << i + 1 << " ";
		cout << endl;
	}
}
